markets/pricing/cli.go

```diff
@@ -0,0 +1,48 @@
+package pricing
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"os/exec"
+
+	"github.com/filecoin-project/go-fil-markets/retrievalmarket"
+	"github.com/filecoin-project/lotus/node/modules/dtypes"
+	"golang.org/x/xerrors"
+)
+
+func CliRetrievalPricingFunc(cmd string) dtypes.RetrievalPricingFunc {
+	return func(ctx context.Context, pricingInput retrievalmarket.PricingInput) (retrievalmarket.Ask, error) {
+		return runPricingFunc(ctx, cmd, pricingInput)
+	}
+}
+
+func runPricingFunc(_ context.Context, cmd string, params interface{}) (retrievalmarket.Ask, error) {
+	j, err := json.Marshal(params)
+	if err != nil {
+		return retrievalmarket.Ask{}, err
+	}
+
+	var out bytes.Buffer
+	var errb bytes.Buffer
+
+	c := exec.Command("sh", "-c", cmd)
```

Comment:
I wonder if we should do payload validation here to defend against potential injection attacks. It's possible that the input is already validated somewhere previous, though. My concern is that we're taking untrusted input (user provided input) and piping it to an shell job...

Reply:
Well, this is coming straight from the miner config, so it's not-so-untrusted user input. Obviously you can still do dumb things with it, but the same can be said about `sudo`

Reply:
I'm not referring to the miner configuration (i.e. the script, or the path to it) -- I'm more referring to the fact that what's being passed as input is a serialized `retrievalmarket.PricingInput`. But that's a constrained type that's populated in our code, i.e. it's not _actually_ user-provided input. So my considerations don't really apply here.

![image](https://user-images.githubusercontent.com/1101242/121670118-00658580-caa5-11eb-8905-06a5d4383271.png)

Reply:
So, this is settled.